### PR TITLE
Add provider to tip compliment

### PR DESCRIPTION
**Pull request by jdkuki**
Fixes brave/brave-browser#4586

Added publisher provider to tweetTip compliment string displayed when sharing a tip via tweet. 
Compliment string now reads "I just tipped [publisher] on [provider]"

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
See steps at brave/brave-browser#4586

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/ui/webui/brave_tip_ui.cc [hunk 1]
```diff
@@ -359,10 +359,13 @@ void RewardsTipDOMHandler::TweetTip(const base::ListValue *args) {
   std::string tweet_id;
   if (!args->GetString(1, &tweet_id))
     return;
+  std::string provider;
+  if (!args->GetString(2, &provider))
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_tip_ui.cc]: Please use this approach `const std::string provider = args->GetList()[0].GetString()`

File: browser/ui/webui/brave_tip_ui.cc [hunk 2]
```diff
@@ -347,22 +347,23 @@ void RewardsTipDOMHandler::OnRecurringTipSaved(
...
   // Retrieve the relevant metadata from arguments.
-  std::string name;
-  if (!args->GetString(0, &name))
-    return;
-  std::string tweet_id;
-  if (!args->GetString(1, &tweet_id))
+  std::string name = args->GetList()[0].GetString();
+  if (name.empty())
     return;
+  std::string tweet_id = args->GetList()[1].GetString();
```
Comment (jdkuki) [inline browser/ui/webui/brave_tip_ui.cc]: `tweet_id` should be allowed to be empty so removed equivalent check used in old API

File: components/resources/brave_components_strings.grd [hunk 3]
```diff
@@ -260,7 +260,7 @@
       <message name="IDS_BRAVE_REWARDS_LOCAL_GRANT_ALREADY_CLAIMED_TEXT" desc="">Sorry, it appears that this grant has already been claimed.</message>
       <message name="IDS_BRAVE_REWARDS_LOCAL_DONAT_NEXT_DATE" desc="Description of the date in the donation box">Next monthly tip date</message>
       <message name="IDS_BRAVE_REWARDS_LOCAL_COMPLIMENT_TWEET" desc="">
-        I just tipped <ph name="NAME">$1<ex>user</ex></ph> using the Brave Browser. Check it out at https://brave.com/tips.
+        I just tipped <ph name="NAME">$1<ex>user</ex></ph> on <ph name="PROVIDER">$2<ex>site</ex></ph> using the Brave Browser. Check it out at https://brave.com/tips.
```
Comment (jdkuki) [inline components/resources/brave_components_strings.grd]: Requires proper formatting, waiting for @NejcZdovc for suggestions.

Comment (jdkuki) [inline components/resources/brave_components_strings.grd]: Requires proper formatting, @NejcZdovc will suggest fix

File: browser/ui/webui/brave_tip_ui.cc (hunk 2 above)
Comment (NejcZdovc) [reply browser/ui/webui/brave_tip_ui.cc]: why should be allowed to be empty?

Comment (jdkuki) [reply browser/ui/webui/brave_tip_ui.cc]: It appears that `tweet_id` is called with an empty parameter in:
https://github.com/brave/brave-core/blob/90b3163772b6df62a2fa7c71343e14b3fee7e5b4/components/brave_rewards/resources/tip/components/tipSite.tsx#L21-L37

I believe the field may be used in the case of a twitter tip to attach the complement as a reply (may be completely wrong) and is not applicable in the case where we tip a site?

Comment (jdkuki) [reply browser/ui/webui/brave_tip_ui.cc]: `tipSite.tsx:35` updated later in this PR

**Comment by NejcZdovc** [comment]
closing as stale, let's revisit when needed

**Pull request closed without merging**